_drafts/practical_zephyr_01_kconfig.md

```diff
@@ -0,0 +1,986 @@
+
+```bash
+$ cat $ZEPHYR_BASE/Kconfig
+# -- hidden comments --
+mainmenu "Zephyr Kernel Configuration"
+source "Kconfig.zephyr"
+```
+
+By sourcing the `Kconfig.zephr` file we're loading all _Kconfig_ menus and symbols provided with Zephr. Next, we declare our own menu between the _"menu"_ and _"endmenu"_ statements to group our application symbols. Within this menu, we declare our `USR_FUN` symbol, which we'll use to enable a function `usr_fun`.
+
+Let's rebuild our application without configuring `USR_FUN` and have a look at the build output:
+
+```bash
+$ west build --board nrf52840dk_nrf52840 -d ../build --pristine
+```
+```
+Parsing /path/to/01_kconfig/Kconfig
+Loaded configuration '/opt/nordic/ncs/v2.4.0/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
+Merged configuration '/path/to/01_kconfig/prj.conf'
+Merged configuration '/path/to/01_kconfig/boards/nrf52840dk_nrf52840.conf'
+Configuration saved to '/path/to/build/zephyr/.config'
+Kconfig
+```
+
+Have a good look at the very first line of the _Kconfig_-related output:
+* In our previous builds, this line indicated the use of Zephyr's default `Kconfig` file as follows:
+  `Parsing /opt/nordic/ncs/v2.4.0/zephyr/Kconfig`,
+* Whereas now it uses our newly created `Kconfig` file:
+  `Parsing /path/to/01_kconfig/Kconfig`.
+
+The remainder of the output should look familiar. When we look at the generated `zephyr/.config` file in our build directory, we see that the `USR_FUN` symbol is indeed set according to its default value `n`:
+
+```bash
+$ cat ../build/zephyr/.config
+```
+```conf
+# Application Options
+#
+# CONFIG_USR_FUN is not set
+# end of Application Options
+```
+
+The new, custom main menu name _"Customized Menu Name"_ is only visible in the interactive _Kconfig_ tools, such as `menuconfig`, and is not used within the generated `zephyr/.config` file. The interactive tools now also show our newly created menu and symbol.
+
+As you can see in the below screenshot, the main menu has the name _"Customized Menu Name"_ and a new menu _"Application Options"_ is now available as the last entry of options:
+
+```bash
+$ west build -d ../build -t menuconfig
+```
+
+![]({% img_url practical-zephyr/kconfig-custom.png %})
+![]({% img_url practical-zephyr/kconfig-application.png %})
+
+> **Notice:** It is also possible to source `Kconfig.zephyr` _after_ defining the application symbols and menus. This would have the effect that your options will be listed _before_ the Zephyr symbols and menus. In this section, we've sourced `Kconfig.zephr` before our own options since this is also the case in the template used by the official documentation.
```

Comment:
Might be the version of docs I'm looking at, but they point to the opposite order where the app configs are defined before `source Kconfig.zephyr`. I would remove the last part of this sentence instead:

```suggestion
> **Notice:** It is also possible to source `Kconfig.zephyr` _after_ defining the application symbols and menus. This would have the effect that your options will be listed _before_ the Zephyr symbols and menus. In this section, we've sourced `Kconfig.zephr` before our own options.
```